### ISSUE TITLE
Operator in OData filter parameter

### ISSUE THREAD

**Issue by camilledervout** [codecutout/StringToExpression#11]
Hi,
I have a problem when i have an operator in a parameter of my filter OData. 
I think one of the solutions could be to indicate that the operator must be preceded and succeeded by a white space. Is this something that I can customize from my side ?

For exemple : "order eq true"
In this case, the parser considers that I am on a "OR" filter and crashes with the following message :

StringToExpression.Exceptions.OperationInvalidException : Unable to perform operation 'der'
        ---- System.ArgumentNullException : Value cannot be null. (Parameter 'member')
      Stack Trace: 
        OperandDefinition.Apply(Token token, ParseState state)
        Parser.Parse(IEnumerable`1 tokens, IEnumerable`1 parameters)
        Language.Parse(String text, ParameterExpression[] parameters)
        ODataFilterLanguage.Parse[T](String text)
        ----- Inner Stack Trace -----
        Expression.MakeMemberAccess(Expression expression, MemberInfo member)
        <>c.<PropertyDefinitions>b__11_1(Expression exp, String prop)
        Enumerable.Aggregate[TSource,TAccumulate](IEnumerable`1 source, TAccumulate seed, Func`3 func)
        <>c.<PropertyDefinitions>b__11_0(String value, ParameterExpression[] parameters)
        OperandDefinition.Apply(Token token, ParseState state)

**Comment by camilledervout** [comment]
Little raise on this subject. 
Another filter problem : moduleName eq ''

StringToExpression.Exceptions.OperationInvalidException:Unabletoperformoperation'uleName'

**Comment by bookwood1977** [comment]
hello, i have similar issue.

there are several solitions:
- 1. write CaseSensitive valid query: - name property exactly like in your entity / model class. . Try "ModuleName eq ..." 

- 2. check the sources; do your own fork, or write a feature branch and do an pull request.
(It seems that package is following ODATA V2  - and so outdated if you need to have ODATA v4)

- There is the class MongoDBFilterLanguage which is a subclass of ODataFilterLanguage (from  https://github.com/codecutout/StringToExpression ) 
- in that class the overridden Function "LogicalOperatorDefinitions"
-- here is the problem... all is similar to OdataFilterLanguage  but the Regular Expressions
-- in short: leading and trailing spaces are removed 
---  ODataFilterLanguage:  regex: @"\b(eq)\b",
--- MongoDBFilterLanguage (this nuget package) :   regex: @"eq",

**Comment by camilledervout** [comment]
For information, the subject is closed : https://github.com/codecutout/StringToExpression/pull/12
The fix is available  in the latest version.
